web/src/lib/components/elements/dropdown.svelte

```diff
@@ -4,13 +4,12 @@
   import { clickOutside } from '$lib/utils/click-outside';
   import { fly } from 'svelte/transition';
   import type Icon from 'svelte-material-icons/DotsVertical.svelte';
+  import { createEventDispatcher } from 'svelte';
 
-  interface DropdownOption {
-    value: string;
-    icon?: Icon;
-  }
-
-  export let options: DropdownOption[] | string[] = [];
+  const dispatch = createEventDispatcher<{
+    changeSort: string;
+  }>();
```

Comment:
Clicking the same option multiple times might not always mean change the sort order. This is a general dropdown component. What about just emitting an `on:select` event. In the parent, if it matches the current value, then you can do something.